### PR TITLE
[Merged by Bors] - feat(Algebraize): Fix algebraize tactic

### PR DESCRIPTION
**Pull request by callesonne**
This fixes algebraize tactic to work better when it is pointing to a (possibly custom) constructor/theorem giving the algebra property from the corresponding ringhom property. As an application we apply this to the new `IsStandardSmooth`, `RingHom` property.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/RingHom/StandardSmooth.lean [hunk 1]
```diff
@@ -31,14 +31,25 @@ namespace RingHom
 variable {R : Type u} {S : Type v} [CommRing R] [CommRing S]
 
 /-- A ring homomorphism `R →+* S` is standard smooth if `S` is standard smooth as `R`-algebra. -/
+@[algebraize RingHom.IsStandardSmooth.toAlgebra]
```
Comment (jcommelin) [inline Mathlib/RingTheory/RingHom/StandardSmooth.lean:34]: Does this mean that `algebraize` does not check that this name exists already?
Should this promise be checked in CI?

File: Mathlib/RingTheory/RingHom/StandardSmooth.lean [hunk 2]
```diff
@@ -31,14 +31,25 @@ namespace RingHom
...
 def IsStandardSmooth (f : R →+* S) : Prop :=
   @Algebra.IsStandardSmooth.{t, w} _ _ _ _ f.toAlgebra
 
+/-- Helper lemma for the `algebraize` tactic.-/
+lemma IsStandardSmooth.toAlgebra {f : R →+* S} (hf : IsStandardSmooth.{t, w} f) :
+    @Algebra.IsStandardSmooth.{t, w} R S _ _ f.toAlgebra := hf
+
 /-- A ring homomorphism `R →+* S` is standard smooth of relative dimension `n` if
 `S` is standard smooth of relative dimension `n` as `R`-algebra. -/
+@[algebraize RingHom.IsStandardSmoothOfRelativeDimension.toAlgebra]
```
Comment (jcommelin) [inline Mathlib/RingTheory/RingHom/StandardSmooth.lean:44]: I think you can get rid of another `letI` on line 57, using this one.

**Comment by callesonne** [comment]
!bench

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**